src/dependency.ts

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable max-classes-per-file */
+
+import { ReplaceSource } from "webpack-sources";
+import * as docGen from "react-docgen-typescript";
+
+// eslint-disable-next-line
+// @ts-ignore: What's the right way to refer to this one?
+import ModuleDependency from "webpack/lib/dependencies/ModuleDependency.js";
+
+// eslint-disable-next-line
+// @ts-ignore: What's the right way to refer to this one?
+import makeSerializable from "webpack/lib/util/makeSerializable.js";
+
+// eslint-disable-next-line
+// @ts-ignore TODO: Figure out where to find a typed version
+import Dependency from "webpack/lib/Dependency.js";
+
+import { generateDocgenCodeBlock } from "./generateDocgenCodeBlock";
+import { LoaderOptions } from "./types";
+
+class DocGenDependency extends ModuleDependency {
+  public static Template: ModuleDependency.Template;
+
+  getReferencedExports(): [] {
+    return Dependency.NO_EXPORTS_REFERENCED;
+  }
+
+  get type(): string {
+    return "docgen";
+  }
+
+  get category(): string {
+    return "docs";
```

Comment:
```suggestion
    return "esm";
```

This affects how this is resolved.

You don't need it for NullDependency